### PR TITLE
Gameselect and room details

### PR DESCRIPTION
**Pull request by carteryu**

### CODE REVIEW

File: civilapp/src/current/RoomDetails.js [hunk 1]
```diff
@@ -196,173 +198,178 @@ class RoomDetails extends React.Component {
...
-          alignItems: 'center',
-          width: '100%',
-        }}
-      >
+      <React.Fragment>
         {this.state.gameSelected ? (
           <View
             style={{
-              flexDirection: 'row',
+              backgroundColor: 'purple',
```
Comment (carteryu) [inline civilapp/src/current/RoomDetails.js]: Need 2 test this view

File: civilapp/src/components/SuperText.js [hunk 2]
```diff
@@ -723,7 +559,7 @@ const RetroBrowseCardText = props => {
     fontFamily = Platform.OS === 'ios' ? 'Upheaval TT (BRK)' : 'upheavtt'
   } else if (props.quarter) {
     color = 'white'
-      ; (fontFamily = 'Sportsball'), (fontSize = 8)
```
Comment (carteryu) [inline civilapp/src/components/SuperText.js]: this is my linter, but whats with the semicolon here?

File: civilapp/src/current/GameSelectScreen.js [hunk 3]
```diff
@@ -55,137 +57,134 @@ class GameSelectScreen extends React.Component {
...
+        return { selectedGameId: 'none' }
+      } else {
+        return { selectedGameId: key }
+      }
     })
+  }
 
-    //This could be useful information to display(not used rn)
-    const channel = gameInfo.Channel
+  // renderStats = (gameInfo, isUpcoming) => {
```
Comment (carteryu) [inline civilapp/src/current/GameSelectScreen.js]: wtf?? i deleted this and itsn ot showing up in git diff

File: civilapp/src/components/SuperText.js (hunk 2 above)
Comment (EphemeralX) [reply civilapp/src/components/SuperText.js]: I haven't seen this kind of code before... should we just assign an object of CSS attributes to a variable and just do that? i.e. `const somethingStyle = { fontFamily: 'Sportsball', fontSize: 8, color: 'white' }`

File: civilapp/src/current/GameSelectScreen.js [hunk 4]
```diff
@@ -237,27 +195,97 @@ class GameSelectScreen extends React.Component {
...
+          message={props.middle}
+        ></RetroSportsBallText>
+      </View>
+    </View>
+    <View
+      style={{
+        flex: 0.4,
+        alignItems: 'flex-end',
+        justifyContent: 'center',
+        //To make up for the fact that shadow is part of the imported asset
```
Comment (EphemeralX) [inline civilapp/src/current/GameSelectScreen.js]: unneeded

File: civilapp/src/current/RoomDetails.js [hunk 5]
```diff
@@ -65,42 +59,45 @@ class RoomDetails extends React.Component {
...
+    const gameDay = gameObj ? gameObj.Day : null
+    const gameTime = gameObj ? gameObj.DateTime : null
 
     this.setState(
       {
         memberList,
         hostInfo,
         onlineMemberList,
+        // onlineUserList,
+        // offlineUserList,
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: are we using this?

File: civilapp/src/current/RoomDetails.js [hunk 6]
```diff
@@ -117,17 +114,47 @@ class RoomDetails extends React.Component {
...
+        if (id !== hostInfo.id) {
+          if (onlineMemberList.find(user => id === user.userId)) {
+            onlineIdList.push(id)
+          } else {
+            offlineIdList.push(id)
+          }
+        }
+      })
+      console.log('op12', onlineIdList)
+      console.log('od123', offlineIdList)
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: rm

File: civilapp/src/current/RoomDetails.js [hunk 7]
```diff
@@ -117,17 +114,47 @@ class RoomDetails extends React.Component {
...
+
+    // })
+
+    /*     memberList.forEach(dbMember => {
       if (onlineMemberList.find(user => dbMember.id === user.userId))
         dbMember.isOnline = true
-    })
-
+    }) */
+    console.log('gay123', offlineUserList)
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: clean up

File: civilapp/src/current/RoomDetails.js [hunk 8]
```diff
@@ -371,341 +423,333 @@ class RoomDetails extends React.Component {
...
+                  width: width * 0.9,
+                }}
+              >
+                <ImageBackground
+                  resizeMode="stretch"
+                  source={BlueButton}
+                  style={{
+                    flex: 1,
+                    width: undefined,
+                    height: height * 0.13, // Im really not sure why the css gods wont let me auto stretch the height of child to parent
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: unneeded comment

File: civilapp/src/current/RoomDetails.js [hunk 9]
```diff
@@ -371,341 +423,333 @@ class RoomDetails extends React.Component {
...
+                  </View>
+                  <FlatList
+                    showsVerticalScrollIndicator={false}
+                    contentContainerStyle={{ marginBottom: '2.5%' }}
+                    data={onlineUserList}
+                    scrollEnabled={false}
+                    keyExtractor={(_, index) => index + ''}
+                    renderItem={({ item }) => {
+                      if (item) {
+                        if (item.id !== roomAdmin) {
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: `if (item && item.id !== roomAdmin) {`

File: civilapp/src/current/RoomDetails.js [hunk 10]
```diff
@@ -371,341 +423,333 @@ class RoomDetails extends React.Component {
...
+                                marginVertical: '2.5%',
+                              }}
+                            ></UserRow>
+                          )
+                        }
+                      }
+                    }}
+                    // ListFooterComponent={
+                    //   !isRoomOwner && isMember ? this.leaveRoomView() : null
+                    // }
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: rm

File: civilapp/src/current/RoomDetails.js [hunk 11]
```diff
@@ -371,341 +423,333 @@ class RoomDetails extends React.Component {
...
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: rm

File: civilapp/src/current/RoomDetails.js [hunk 12]
```diff
@@ -371,341 +423,333 @@ class RoomDetails extends React.Component {
...
-    fontFamily: 'Sportsball',
-    color: '#FFFFFF',
-    textAlign: 'left',
-    marginTop: '2.5%',
-    marginLeft: LeftwardMargin,
-  },
-})
+const CustomNoViewBar = () => {
+  return <View style={{ padding: 0 }} />
+}
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDetails.js]: why do we need this/why are we assigning this to a variable here?

File: civilapp/src/current/RoomDrawer.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import RoomDetails from './RoomDetails.js'
+import { withNavigation } from 'react-navigation'
+class RoomDrawer extends React.Component {
+  constructor(props) {
+    super(props)
+  }
+  render() {
+    return <RoomDetails></RoomDetails>
```
Comment (EphemeralX) [inline civilapp/src/current/RoomDrawer.js]: <RoomDetails/>

File: civilapp/src/current/RoomViewUtils/index.js [hunk 14]
```diff
@@ -21,8 +21,26 @@ const setOnRemoveSpeakerSub = async (id, callback) => {
...
+  }
+  if (minutesRemaining && minutesRemaining > 0) {
+    finalDisplayTime = minutesRemaining.toString() + finalDisplayTime
+  } else {
+    finalDisplayTime = '0' + finalDisplayTime
+  }
+  return finalDisplayTime
+}
+
+export { setOnAddSpeakerSub, setOnRemoveSpeakerSub, displayTime }
```
Comment (EphemeralX) [inline civilapp/src/current/RoomViewUtils/index.js]: I think we should remove this index.js and refactor these into multiple utility classes.  maybe a TimeUtil for you, and a SubscriptionsUtil for Pedro who put this here.

File: civilapp/src/profile/ProfileScreenNew.js [hunk 15]
```diff
@@ -83,26 +75,6 @@ class ProfileScreenNew extends React.Component {
       </TouchableOpacity>
     )
 
-    logOut = async () => {
```
Comment (EphemeralX) [inline civilapp/src/profile/ProfileScreenNew.js]: it's fine to remove this in this file, just make sure that there are no related references to this in this file (github makes it hard to check this in a PR)

Comment (EphemeralX) [reply civilapp/src/profile/ProfileScreenNew.js]: +remove unused imports, styling in addition to code

File: civilapp/src/current/GameSelectScreen.js (hunk 4 above)
Comment (carteryu) [reply civilapp/src/current/GameSelectScreen.js]: I think the comment is useful

File: civilapp/src/current/RoomDetails.js (hunk 12 above)
Comment (carteryu) [reply civilapp/src/current/RoomDetails.js]: its how the library works + you can customize it later if need be

File: civilapp/src/current/RoomDrawer.js (hunk 13 above)
Comment (carteryu) [reply civilapp/src/current/RoomDrawer.js]: lol aitee

File: civilapp/src/current/RoomDetails.js (hunk 9 above)
Comment (carteryu) [reply civilapp/src/current/RoomDetails.js]: hm does it check if item is good first and THEN check item.id?

File: civilapp/src/current/RoomDetails.js (hunk 8 above)
Comment (carteryu) [reply civilapp/src/current/RoomDetails.js]: ok rmd

File: civilapp/src/profile/ProfileScreenNew.js (hunk 15 above)
Comment (carteryu) [reply civilapp/src/profile/ProfileScreenNew.js]: this file has no unused imports and wdym styling in addition to code?

Comment (carteryu) [reply civilapp/src/profile/ProfileScreenNew.js]: and no related references

File: civilapp/src/components/SuperText.js (hunk 2 above)
Comment (carteryu) [reply civilapp/src/components/SuperText.js]: i dont knwo what the code does i was just confused about the semi colon

File: civilapp/src/current/RoomViewUtils/index.js (hunk 14 above)
Comment (carteryu) [reply civilapp/src/current/RoomViewUtils/index.js]: sure later. will be useful when utils file gets more bloated

**Pull request merged**